sebs/benchmark.py

```
@@ -298,283 +426,378 @@ def query_cache(self):
+            else: # Fallback for environments where mounts are problematic (e.g., CI)
+                self.logging.info("Using Docker cp for file transfer due to 'docker_copy_build_files' flag.")
+                container = self._docker_client.containers.run(
+                    f"{repo_name}:{final_image_name_to_use}",
+                    environment={"APP": self.benchmark, 
+                                 "TARGET_ARCHITECTURE": self._experiment_config.architecture},
+                    user=container_user_id, # Run as current host user if possible
+                    remove=True, detach=True, tty=True, command="/bin/bash", # Keep alive
+                )
+                # Copy files to container
+                tar_archive_path = os.path.join(os.path.dirname(output_dir), "function_package.tar")
+                with tarfile.open(tar_archive_path, "w") as tar:
```

Ruff (0.11.9)

Undefined name `tarfile`

(F821)